planet/src/main/webapp/js/profile-events-renderer.js

```diff
@@ -0,0 +1,257 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+let userId;
+let listIndex = 0;
+let eventsList = [];
+let listName = 'List X';
+
+export const ProfileEventsRenderer = { 
+    init: async (uid) => {
```

Comment:
The init pattern is generally frowned upon since there's no guarantee that init() will be called.
Refactor this into an actual class that takes a user id (along with other required fields) in its constructor.

Reply:
Actually, this is just an object with a function that I called init to render the initial list of events. The function gets called in the profile.js script. I think I should probably change its name to something less confusing. 
I don't really need a class but just a module and the reason I am putting the function inside the object is because it allows me to import everything i want to import from this module at once (functions and/or variables).